extensions/cpsection/language/model.py

```diff
@@ -81,49 +81,54 @@ def _initialize():
 
 
 def _write_i18n(lang_env, language_env):
-    path = os.path.join(os.environ.get('HOME'), '.i18n')
-    if not os.access(path, os.W_OK):
-        print _standard_msg
-        fd = open(path, 'w')
-        fd.write('LANG="%s"\n' % _default_lang)
-        fd.write('LANGUAGE="%s"\n' % _default_lang)
-        fd.close()
-    else:
-        fd = open(path, 'w')
-        fd.write('LANG="%s"\n' % lang_env)
-        fd.write('LANGUAGE="%s"\n' % language_env)
-        fd.close()
+    path = os.path.join(os.environ.get('HOME', ''), '.i18n')
+    # try avoid writing the file if the values didn't changed
+    lang_line = None
+    language_line = None
+    try:
+        with open(path) as fd:
+            for line in fd:
+                if line.startswith('LANG='):
+                    lang_line = line
+                elif line.startswith('LANGUAGE='):
+                    language_line = line
+    except:
+        pass
+
+    new_lang_line = 'LANG="%s"\n' % lang_env
+    new_language_line = 'LANGUAGE="%s"\n' % language_env
+
+    if lang_line == new_lang_line and language_line == new_language_line:
+        return
 
+    try:
+        with open(path, 'w') as fd:
+            fd.write(new_lang_line)
```

Comment:
Lines 105 to 110 do not preserve previous unrelated contents of file.  Since the file can contain locale environment variables like currency and date format, this is impolite to other software. :smile: